core/client/src/main/java/alluxio/client/block/stream/LocalFilePacketWriter.java

```diff
@@ -70,9 +70,10 @@ public int packetSize() {
   public void writePacket(final ByteBuf buf) throws IOException {
     try {
       Preconditions.checkState(!mClosed, "PacketWriter is closed while writing packets.");
-      ensureReserved(mPos + buf.readableBytes());
-      mPos += buf.readableBytes();
-      buf.readBytes(mWriter.getChannel(), buf.readableBytes());
+      int sz = buf.readableBytes();
+      ensureReserved(mPos + sz);
+      mPos += sz;
+      Preconditions.checkState(buf.readBytes(mWriter.getChannel(), sz) == sz);
```

Comment:
similarly, i think this should be cheap. I'd prefer to keeping this one to check the return value of buf.readBytes